skimage/util/_coord_conv.py

```diff
@@ -0,0 +1,234 @@
+import numpy as np
+
+
+def xy_to_rc(points=None, image=None):
+    """Convert points' coordinates and/or image from `x, y` to `r, c` convention.
+
+        o--->....       o--->....
+        |   x   .       |   c   .
+        |       .  ==>  |       .
+        v y     .       v r     .
+        .........       .........
+
+    Parameters
+    ----------
+    points : (K, 2) ndarray, optional
+        Array of points' `x, y` coordinates.
+    image : (M, N[, C]) ndarray, optional
+        Image stored in `x, y` axes convention.
+
+    Returns
+    -------
+    points_out : (K, 2) ndarray
+        Array of `points` `r, c` coordinates.
+    image_out : (M, N[, C]) ndarray
+        Same data as `image`, but in `r, c` convention.
+    """
+    if points is not None:
+        points_out = np.fliplr(points)
+    else:
+        points_out = None
+
+    if image is not None:
+        image_out = np.swapaxes(image, 0, 1)
+    else:
+        image_out = None
+
+    return points_out, image_out
+
+
+def rc_to_xy(points=None, image=None):
+    """Convert points' coordinates and/or image from `r, c` to `x, y` convention.
+
+        o--->....       o--->....
+        |   c   .       |   x   .
+        |       .  ==>  |       .
+        v r     .       v y     .
+        .........       .........
+
+    Parameters
+    ----------
+    points : (K, 2) ndarray, optional
+        Array of points' `r, c` coordinates.
+    image : (M, N[, C]) ndarray, optional
+        Image stored in `r, c` axes convention.
+
+    Returns
+    -------
+    points_out : (K, 2) ndarray
+        Array of `points` `x, y` coordinates.
+    image_out : (M, N[, C]) ndarray
+        Same as `image`, but in `x, y` convention.
+    """
+    if points is not None:
+        points_out = np.fliplr(points)
+    else:
+        points_out = None
+
+    if image is not None:
+        image_out = np.swapaxes(image, 0, 1)
+    else:
+        image_out = None
+
+    return points_out, image_out
```

Comment:
I was thinking about this... It is probably a matter of balance: nested function call requires some extra resources. These functions are supposed to be used quiet often, so less operations is better.
You may have noticed, that I do not check if `points` are within `image` frame. Exactly for the same reason.
Another story is the two last functions, which I think will be used less frequently, so I use function chaining there  to avoid duplication.